src/components/File.vue

```diff
@@ -21,108 +21,203 @@
  -->
 
 <template>
-	<a :class="{
-			'file--cropped': croppedLayout,
-		}"
-		class="file"
-		:href="davPath"
-		:aria-label="ariaLabel"
-		@click.prevent="openViewer">
-		<div v-if="item.injected.mime.includes('video') && item.injected.hasPreview" class="icon-video-white" />
-		<!-- image and loading placeholder -->
-		<transition-group name="fade" class="transition-group">
-			<img v-if="!error"
-				ref="img"
-				:key="`${item.injected.basename}-img`"
-				:src="src"
-				:alt="item.injected.basename"
-				:aria-describedby="ariaUuid"
-				@load="onLoad"
-				@error="onError">
-
-			<svg v-if="!loaded || error"
-				:key="`${item.injected.basename}-svg`"
-				xmlns="http://www.w3.org/2000/svg"
-				viewBox="0 0 32 32"
-				fill="url(#placeholder__gradient)">
-				<use v-if="isImage" href="#placeholder--img" />
-				<use v-else href="#placeholder--video" />
-			</svg>
-		</transition-group>
-
-		<!-- image name and cover -->
-		<p :id="ariaUuid" class="hidden-visually">{{ item.injected.basename }}</p>
-		<div class="cover" role="none" />
-	</a>
+	<div class="file-container"
+		:class="{selected}">
+		<CheckboxRadioSwitch v-if="allowSelection"
+			class="selection-checkbox"
+			:checked="selected"
+			@update:checked="onToggle" />
+
+		<a class="file"
+			:href="davPath"
+			:aria-label="ariaLabel"
+			@click.prevent="emitClick">
+
+			<div v-if="item.mime.includes('video') && item.hasPreview" class="icon-video-white" />
+
+			<!-- image and loading placeholder -->
+			<div class="images-container">
+				<img v-if="visibility !== 'none' && canLoad && !error"
+					ref="imgNear"
+					:key="`${item.basename}-near`"
+					:src="srcNear"
+					:alt="item.basename"
+					:aria-describedby="ariaUuid"
```

Comment:
```suggestion
					:aria-describedby="ariaId"
```
**suggestion (blocking):** If `fileid` is not a UUID compliant string this should be renamed to something like `ariaId`

Reply:
Nice catch, I renamed it to ariaDescription for even more clarity.